src/Explorer/Controls/Settings/SettingsComponent.tsx

```diff
@@ -0,0 +1,2588 @@
+import TriangleRight from "../../../../images/Triangle-right.svg";
+import TriangleDown from "../../../../images/Triangle-down.svg";
+import InfoBubble from "../../../../images/info-bubble.svg";
+import TelemetryProcessor from "../../../Shared/Telemetry/TelemetryProcessor";
+import { Action } from "../../../Shared/Telemetry/TelemetryConstants";
+import { PlatformType } from "../../../PlatformType";
+import { RequestOptions } from "@azure/cosmos/dist-esm";
+import Explorer from "../../Explorer";
+import { updateOffer } from "../../../Common/DocumentClientUtilityBase";
+import { updateCollection } from "../../../Common/dataAccess/updateCollection";
+import { CommandButtonComponentProps } from "../../Controls/CommandButton/CommandButtonComponent";
+import { userContext } from "../../../UserContext";
+import { updateOfferThroughputBeyondLimit } from "../../../Common/dataAccess/updateOfferThroughputBeyondLimit";
+import SettingsTab from "../../Tabs/SettingsTabV2";
+import { ThroughputInputAutoPilotV3Component } from "../ThroughputInput/ThroughputInputReactComponentAutoPilotV3";
+import { ThroughputInputComponent } from "../ThroughputInput/ThroughputInputReactComponent";
+import {
+  StatefulValuesType,
+  StatefulValue,
+  isDirty,
+  getAutoPilotV3SpendElement,
+  getAutoPilotV2SpendElement,
+  getEstimatedSpendElement,
+  getEstimatedAutoscaleSpendElement,
+  manualToAutoscaleDisclaimerElement,
+  ttlWarning,
+  indexingPolicyTTLWarningMessage,
+  updateThroughputBeyondLimitWarningMessage,
+  updateThroughputDelayedApplyWarningMessage,
+  getThroughputApplyDelayedMessage,
+  getThroughputApplyShortDelayMessage,
+  getThroughputApplyLongDelayMessage
+} from "./SettingsUtils";
+import { AccessibleElement } from "../AccessibleElement/AccessibleElement";
+
+enum ChangeFeedPolicyToggledState {
+  Off = "Off",
+  On = "On"
+}
+
+enum TtlType {
+  Off = "off",
+  On = "on",
+  OnNoDefault = "on-nodefault"
+}
+
+enum GeospatialConfigType {
+  Geography = "Geography",
+  Geometry = "Geometry"
+}
+
+interface ButtonV2 {
+  isVisible: () => boolean;
+  isEnabled: () => boolean;
+  isSelected?: () => boolean;
+}
+
+enum StatefulValueNames {
+  Throughput = "throughput",
+  TimeToLive = "timeToLive",
+  TimeToLiveSeconds = "timeToLiveSeconds",
+  GeospatialConfigType = "geospatialConfigType",
+  IndexingPolicyContent = "indexingPolicyContent",
+  Rupm = "rupm",
+  ConflictResolutionPolicyMode = "conflictResolutionPolicyMode",
+  ConflictResolutionPolicyPath = "conflictResolutionPolicyPath",
+  ConflictResolutionPolicyProcedure = "conflictResolutionPolicyProcedure",
+  AnalyticalStorageTtlSelection = "analyticalStorageTtlSelection",
+  AnalyticalStorageTtlSeconds = "analyticalStorageTtlSeconds",
+  ChangeFeedPolicyToggled = "changeFeedPolicyToggled",
+  AutoPilotThroughput = "autoPilotThroughput"
+}
+
+interface UpdateStatefulValueParams {
+  key: keyof SettingsComponentState;
+  value: StatefulValuesType;
+  updateCurrent: boolean;
```

Comment:
I'm not seeing a place where we use `updateCurrent: false`. Is it needed?

Reply:
good catch. there is no scenario in which we would update the baseline and not update current value. WIll remove this flag.